vmdb/app/models/storage.rb

```diff
@@ -360,24 +360,24 @@ def self.scan_timer(zone_name = nil)
 
   def scan(userid = "system", role = "ems_operations")
     log_header = "MIQ(Storage.scan)"
-    raise(MiqException::MiqUnsupportedStorage, "Action not supported for #{ui_lookup(:table=>"storages")} type [#{self.store_type}], [#{self.name}] with id: [#{self.id}]") unless SUPPORTED_STORAGE_TYPES.include?(self.store_type)
+    raise(MiqException::MiqUnsupportedStorage, "Action not supported for #{ui_lookup(:table => "storages")} type [#{self.store_type}], [#{self.name}] with id: [#{self.id}]") unless SUPPORTED_STORAGE_TYPES.include?(self.store_type)
 
     hosts = self.active_hosts_with_credentials
-    raise(MiqException::MiqStorageError,       "Check that a Host is running and has valid credentials for #{ui_lookup(:table=>"storages")} [#{self.name}] with id: [#{self.id}]") if hosts.empty?
+    raise(MiqException::MiqStorageError,       "Check that a Host is running and has valid credentials for #{ui_lookup(:table => "storages")} [#{self.name}] with id: [#{self.id}]") if hosts.empty?
 
     task_name = "SmartState Analysis for [#{self.name}]"
     self.class.create_scan_task(task_name, userid, [self])
   end
 
   def self.unregistered_vm_config_files
-    Storage.find(:all).inject([]) {|list, s| list + s.unregistered_vm_config_files}
+    Storage.all.inject([]) { |list, s| list + s.unregistered_vm_config_files }
   end
 
   def unmanaged_vm_config_files
     files = if association_cache.include?(:storage_files)
       self.storage_files.select { |f| f.ext_name == "vmx" && f.vm_or_template_id.nil? }
     else
-      self.storage_files.all(:conditions => {:ext_name => "vmx", :vm_or_template_id => nil})
+      self.storage_files.where(:ext_name => "vmx", :vm_or_template_id => nil)
```

Comment:
Then again maybe not...I think the idea was to prevent AR from going back to the database all the time, and calling storage_files caches it once, so the other methods (like `managed_vm_config_files`) can just do an in-memory select on the cache.